File: ex412/go/main.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	fmt.Println(fizzBuzz(15))
}

func fizzBuzz(n int) []string {
	lista := make([]string, n)

	for i := range lista {
		value := i + 1
		if value%3 == 0 && value%5 == 0 {
			lista[i] = "FizzBuzz"
		} else if value%3 == 0 {
			lista[i] = "Fizz"
		} else if value%5 == 0 {
			lista[i] = "Buzz"
		} else {
			lista[i] = strconv.Itoa(value)
		}
	}
	return lista
}

File: ex412/java/Main.java
package ex412.java;

import java.util.ArrayList;
import java.util.List;

public class Main {
    public static void main(String[] args) {
        System.out.println(fizzFizz(5));
    }

    public static List<String> fizzFizz(int n) {
        List<String> answer = new ArrayList<>(n);

        for (int i = 1; i <= n; i++) {
            boolean divisibleBy3 = i % 3 == 0;
            boolean divisibleBy5 = i % 5 == 0;

            if (divisibleBy3 && divisibleBy5) {
                answer.add("FizzBuzz");
            } else if (divisibleBy3) {
                answer.add("Buzz");
            } else if (divisibleBy5) {
                answer.add("Fizz");
            } else {
                answer.add(String.valueOf(i));
            }
        }
        return answer;
    }
}